### PR TITLE
Add LGTM code quality badges

### PR DESCRIPTION
**Pull request by xcorail**
Hi there!

I thought you might be interested in adding these [code quality badges](https://lgtm.com/projects/g/apache/calcite/ci/#badges) to your project. They will indicate a very high code quality to potential users and contributors.
You can also check the alerts discovered by [LGTM](https://lgtm.com/projects/g/apache/calcite).

N.B.: I am on the team behind LGTM.com, I'd appreciate your feedback on this initiative, whether you're interested or not, if you find time to drop me a line. Thanks.

### CODE REVIEW

**Comment by vlsi** [comment]
@xcorail , that looks nice.

Just wondering, does LGTM integrate with GitHub checks?

**Comment by vlsi** [comment]
I meant something like https://developer.github.com/changes/2018-05-07-new-checks-api-public-beta/ 
It would be great if LGTM could add comments next to the changed lines (like a person would do).

**Comment by xcorail** [comment]
Ha ok
Let me rephrase to make sure I get your point: 
At the moment, you'll get a report saying for example 1 new alert, and when you click on the report, it will open a LGTM window, and you'll see the LGTM comments next to the relevant changed lines but in this other window
With the GitHub check integration, you would see the LGTM comments directly in the native diff. 
It is interesting indeed, I will pass this idea to the team!

**Comment by sjvs** [comment]
Hi @vlsi! Thanks for your suggestion. We're currently working on integrating the new GitHub Apps APIs into LGTM; we'll be working on the checks API after that. Give us a shout if you have any other suggestions/comments/questions!

**Comment by xcorail** [comment]
Hey @vlsi 

Just wanted to let you know that during 6 weeks, for each LGTM alert fixed, we'll make a donation to WWF. You can also win a free ticket to GitHub Universe, with travel and accommodation. Details here: https://competitions.lgtm.com/ghu-2018

**Comment by risdenk** [comment]
Not sure how I feel about this one. Based on https://issues.apache.org/jira/browse/INFRA-17226 LGTM can't be integrated with the build. Since INFRA is unable to add LGTM to builds due to repository access requirements, that would make this like an advertisement that we aren't actually using.

**Comment by julianhyde** [comment]
I'm -0 on this. Doesn't seem to be worth the effort.

The PR submitter works for the company, so we would be free advertising for them. But our labor is not free.

**Comment by risdenk** [comment]
Closing this PR due to no immediate need to move forward. If LGTM and Apache legal due to Github permissions end up working together we can look at this again.
